crates/compiler/gen_dev/src/generic64/aarch64.rs

```diff
@@ -919,17 +980,14 @@ impl AArch64CallStoreArgs {
                 lambda_set.runtime_representation(),
             ),
             LayoutRepr::Struct { .. } => {
-                let stack_offset = self.tmp_stack_offset;
-
-                let size = copy_symbol_to_stack_offset::<CC>(
-                    buf,
-                    storage_manager,
-                    sym,
-                    tmp_reg,
-                    stack_offset,
-                );
-
-                self.tmp_stack_offset += size as i32;
+                let stack_size = layout_interner.stack_size(in_layout);
+                if stack_size <= 8 {
+                    self.store_arg_64bit(buf, storage_manager, sym);
+                } else if stack_size <= 16 {
+                    self.store_arg_128bit(buf, storage_manager, sym);
+                } else {
+                    unreachable!("covered by earlier branch");
+                }
```

Comment:
now structs that are small enough are passed via registers. I've also added this for the systemv calling convention which does this in the same way. Windows will need independent validation, but is not tested on CI right now.